test/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring.Tests/CalculatorTests.cs

```diff
@@ -192,4 +193,29 @@ public void OverUtilized()
         Assert.Equal(1500UL, record.MemoryUsedInBytes);
         Assert.Equal(1000UL, record.SystemResources.GuaranteedMemoryInBytes);
     }
+
+    /// <summary>
```

Comment:
This can be combined with the test above, for example:

```
    [Theory]
    [InlineData(1, 100.0)]
    [InlineData(2, 100.0)]
    public void OverUtilized(double cpuUnits, double expectedCpuUsage)
    {
        var limitedResources = new SystemResources(cpuUnits, cpuUnits, TotalMemoryInBytes, TotalMemoryInBytes);
        var secondSnapshotTimeSpan = _firstSnapshot.TotalTimeSinceStart.Add(TimeSpan.FromSeconds(5));

        // Now, what's the total number of available ticks between the two samples.
        var totalAvailableTicks = secondSnapshotTimeSpan.Ticks - _firstSnapshot.TotalTimeSinceStart.Ticks;

        var secondSnapshot = new Snapshot(
            totalTimeSinceStart: secondSnapshotTimeSpan,

            // Set each of kernel and uesr time to all the available ticks between
            // the 2 snapshots to make them sum to 200% of the total CPU time.
            kernelTimeSinceStart: TimeSpan.FromTicks(totalAvailableTicks),
            userTimeSinceStart: TimeSpan.FromTicks(totalAvailableTicks),
            memoryUsageInBytes: 1500);

        // Now, when we run the calculator, CPU should be at 100%.
        var record = Calculator.CalculateUtilization(_firstSnapshot, secondSnapshot, limitedResources);
        Assert.Equal(expectedCpuUsage, record.CpuUsedPercentage);

        // Assert that memory is at 100%
        Assert.Equal(100.0, record.MemoryUsedPercentage);
        Assert.Equal(1500UL, record.MemoryUsedInBytes);
        Assert.Equal(1000UL, record.SystemResources.GuaranteedMemoryInBytes);
    }
```

Reply:
Combined 2 tests